xmount: add page (#6106)
tldr-pages/tldr: pages/linux/xmount.md

# xmount

> > Convert on-the-fly between multiple input and output harddisk image types with optional write cache support.

Reviewer: ```suggestion
> Convert on-the-fly between multiple input and output hard disk image types with optional write cache support.
```

> > Allows you to convert on-the-fly between multiple input and output harddisk image types with optional write cache support.

Reviewer 2: Slightly more concise:
```suggestion
> Convert on-the-fly between multiple input and output harddisk image types with optional write cache support.
```

> Creates a virtual file system using FUSE (Filesystem in Userspace) that contains a virtual representation of the input image.

> > More information: <https://www.pinguin.lu/xmount>.

Reviewer 3: ```suggestion
> More information: <https://manned.org/xmount>.
```
Maybe we should use this link, since the official page doesn't really provide any useful information

> - Mount a RAW image file into a DMG container file:

Reviewer 3: ```suggestion
- Mount a `.raw` image file into a DMG container file:
```
I know this breaks the uniformity, but I think otherwise it can be read as `mount a raw image file` too easily.

> - Mount an RAW-Imagefile into a DMG-container file:
> 
> `xmount --in raw IMAGE.dd --out dmg MOUNTPOINT`

Reviewer 2: I'd rearrange the command this way:
```suggestion
- Mount a RAW image file into a DMG container file:

`xmount --in {{raw}} --out {{dmg}} {{path/to/image.dd}} {{mountpoint}}`
```
Because having `raw` right before the `.dd` file, and `dmg` right before `mointpoint` makes seem as though the `.dd` is part of `--in` and `mountpoint` is part of `--out`, when in reality they the files have nothing to do with the flags — the flags merely indicate the input/output file _types_.

Author: I like the suggestion with the {{ bracklets but we have to change the order because the description of the format is part  of the syntax --in or --out.

Your syntax would trigger this warning on execution:
_WARNING: ParseCmdLine@711 : You are using a deprecated --in option syntax which will be removed in the next release. Please see the man page on how to use the new syntax._
```suggestion
- Mount a RAW image file into a DMG container file:

`xmount --in {{raw}} {{path/to/image.dd}} --out {{dmg}} {{mountpoint}}`
```

> - Mount an EWF-Imagefile with write-cache support into a VHD file to boot from:
> 
> `xmount --cache CACHE.ovl --in ewf IMAGE.E?? --out vhd MOUNTPOINT`

Reviewer 2: ```suggestion
- Mount an EWF image file with write-cache support into a VHD file to boot from:

`xmount --cache {{path/to/cache.ovl}} --in {{ewf}} --out {{vhd}} {{path/to/image.E??}} {{mountpoint}}`
```

Author: Same problem here with the order.

```suggestion
- Mount an EWF image file with write-cache support into a VHD file to boot from:

`xmount --cache {{path/to/cache.ovl}} --in {{ewf}} {{path/to/image.E??}} --out {{vhd}} {{mountpoint}}`
```

> `xmount --cache {{path/to/cache.ovl}} --in {{ewf}} {{path/to/image.E??}} --out {{vhd}} {{mountpoint}}`

Reviewer 3: What is `E??`?

Reviewer 2: I have no idea, but it's used in the manpage so it seems fine to me.

> - Mount first partition at sector 2048 into a new RAW image file:

Reviewer 3: ```suggestion
- Mount the first partition at sector 2048 into a new RAW image file:
```
I'm not really understanding this command. The first partition on which filesystem?

> - Mount the first partition at sector 2048 into a new RAW image file:

Reviewer 2: Per @marchersimon's other suggestion
```suggestion
- Mount the first partition at sector 2048 into a new `.raw` image file:
```

> - Mount first partition at sector 2048 into a new RAW-Image:
> 
> `xmount --offset 2048 --in raw IMAGE.dd --out  raw MOUNTPOINT`

Reviewer 2: ```suggestion
- Mount first partition at sector 2048 into a new RAW image file:

`xmount --offset {{2048}} --in {{raw}} --out {{raw}} {{path/to/image.dd}} {{mountpoint}}`
```

Author: ```suggestion
- Mount first partition at sector 2048 into a new RAW image file:

`xmount --offset {{2048}} --in {{raw}} {{path/to/image.dd}} --out {{raw}} {{mountpoint}}`
```

# xmount

> Allows you to convert on-the-fly between multiple input and output harddisk image types with optional write cache support.
> Creates a virtual file system using FUSE (Filesystem in Userspace) that contains a virtual representation of the input image.

> 

Reviewer 2: ```suggestion
```
